File: src/colors.rs
//! Color constants.

pub const BACKGROUND: [f32; 3] = [0.3, 0.3, 0.3];

pub const OUTLINE_BLACK: [f32; 4] = [0.0, 0.0, 0.0, 1.0];
pub const OUTLINE_WHITE: [f32; 4] = [1.0, 1.0, 1.0, 1.0];

pub const RED: [f32; 3] = [0.75, 0.0, 0.05];
pub const ORANGE: [f32; 3] = [0.75, 0.4, 0.0];
pub const WHITE: [f32; 3] = [0.75, 0.75, 0.75];
pub const YELLOW: [f32; 3] = [0.65, 0.75, 0.2];
pub const GREEN: [f32; 3] = [0.0, 0.6, 0.0];
pub const BLUE: [f32; 3] = [0.0, 0.3, 0.75];
pub const PURPLE: [f32; 3] = [0.5, 0.2, 0.75];
pub const PINK: [f32; 3] = [0.75, 0.45, 0.6];

pub const LABEL_FG: [f32; 4] = [1.0, 1.0, 1.0, 1.0];
pub const LABEL_BG: [f32; 4] = [0.0, 0.0, 0.0, 1.0];

/// Returns the background color.
pub fn get_bg() -> (f32, f32, f32, f32) {
    let [r, g, b] = BACKGROUND;
    (r, g, b, 1.0)
}

File: lua/common/colors.lua
local function add_symmetric_colors(...)
  for v in ... do
    add_color(v)
  end
end

function cube(radius)
  add_symmetric_colors(cd{4, 3}:expand('oox'))
  name_colors{'Right', 'Up', 'Left', 'Down', 'Front', 'Back'}
  reorder_colors{'Right', 'Left', 'Up', 'Down', 'Front', 'Back'}
  set_default_colors{'red', 'orange', 'white', 'yellow', 'green', 'blue'}
end

function dodecahedron(radius)
  add_symmetric_colors(cd{5, 3}:expand('oox'))
end

function hypercube(radius)
  add_symmetric_colors(cd{4, 3, 3}:expand('ooox'))
  name_colors{'Right', 'Up', 'Left', 'Down', 'Front', 'Back', 'Out', 'In'}
  reorder_colors{'Right', 'Left', 'Up', 'Down', 'Front', 'Back', 'Out', 'In'}
  set_default_colors{'red', 'orange', 'white', 'yellow', 'green', 'blue', 'pink', 'purple'}
end
